pkg/api/v1/atlasnetworkpeering_types.go

```diff
@@ -0,0 +1,86 @@
+/*
+Copyright 2020 MongoDB.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import "github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/api/v1/provider"
+
+// +k8s:deepcopy-gen=false
+
+type AtlasNetworkPeeringConfig struct {
+	//ProviderName is the name of the provider. If not set, it will be set to "AWS".
+	// +optional
+	ProviderName provider.ProviderName `json:"providerName,omitempty"`
+	// ID of the network peer container. If not set, operator will create a new container with ContainerRegion and AtlasCIDRBlock input.
+	// +optional
+	ContainerID string `json:"containerId"`
+
+	// AWSConfiguration is the specific AWS settings for network peering
+	// +kubebuilder:validation:Optional
+	AWSConfiguration *AWSNetworkPeeringConfiguration `json:"awsConfiguration,omitempty"`
+	// AzureConfiguration is the specific Azure settings for network peering
+	// +kubebuilder:validation:Optional
+	AzureConfiguration *AzureNetworkPeeringConfiguration `json:"azureConfiguration,omitempty"`
+	// GCPConfiguration is the specific Google Cloud settings for network peering
+	// +kubebuilder:validation:Optional
+	GCPConfiguration *GCPNetworkPeeringConfiguration `json:"gcpConfiguration,omitempty"`
+}
+
+// +k8s:deepcopy-gen=false
+
+type AtlasProviderContainerConfig struct {
+	// ContainerRegion is the provider region name of Atlas network peer container. If not set, AccepterRegionName is used.
+	// +optional
+	ContainerRegion string `json:"containerRegion"`
+	//Atlas CIDR. It needs to be set if ContainerID is not set.
+	// +optional
+	AtlasCIDRBlock string `json:"atlasCidrBlock"`
+}
+
+// +k8s:deepcopy-gen=false
```

Comment:
why is this omitting deepgen?

Reply:
not a CRD yet, explained elswhere

Reply:
deepcopy is important (maybe not here but elsewhere) in the translation. I _think_ it is possible to generate deepcopy without having it CRD generated. Try not using the Scheme registration, maybe that is enough.

Reply:
and in fact, deepcopy is valuable for translation layer types as well.

Reply:
will try to see if we can use deep copy while not making a CRD yet.

Reply:
removed and all is fine, I think I rejected the changes to deep copy autogen, but you are right they are already useful even before the CRD

Reply:
did you push all commits? I still see it here.

Reply:
not yet, but I already tested locally